evm/gateway.js

```diff
@@ -318,7 +318,16 @@ async function processCommand(config, chain, options) {
         }
 
         case 'isContractCallApproved': {
-            const { commandId, destination, payloadHash, sourceChain, sourceAddress } = options;
+            const { destination, payloadHash, sourceChain, sourceAddress } = options;
+
+            let commandId;
+
+            if (options.messageId) {
+                // Derive commandId for Amplifier gateway
+                commandId = id(`${sourceChain}_${options.messageId}`);
+            } else {
+                commandId = options.commandID.startsWith('0x') ? options.commandID : id(parseInt(options.commandID).toString());
```

Comment:
why is an integer commandId being passed in? only for testing? not very obvious, should add comments. Another option is to pad the 0x value to make it 32 bytes so the same format can be used

Reply:
commandId failed to pass validation check when calling function here hence the conversion. This is only for testing.

Reply:
can we revert the change then? There is also an `isMessageApproved` method that can be used for amplifier EVM chains

Reply:
Revert the change